tests/requirements_test.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2016 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for perfkitbenchmarker.requirements."""
+
+import contextlib
+import os
+import StringIO
+import unittest
+
+import mock
+
+from perfkitbenchmarker import errors
+from perfkitbenchmarker import requirements
+
+
+_BRANCH_ROOT_DIR = os.path.dirname(os.path.dirname(__file__))
+_PATH = 'path'
+
+
+class CheckRequirementsTestCase(unittest.TestCase):
+
+  @contextlib.contextmanager
+  def _MockOpen(self, file_content):
+    io = StringIO.StringIO(file_content)
+    with mock.patch(requirements.__name__ + '.open', return_value=io):
+      yield
+
+  def testFulfilledRequirement(self):
+    requirements_content = """
+    # Comment line, blank line, and a fulfilled requirement.
+
+    python-gflags>=2.0
+    """
+    with self._MockOpen(requirements_content):
+      requirements._CheckRequirements(_PATH)
+
+  def testMissingPackage(self):
+    requirements_content = """
+    # This is not a real package.
+    perfkitbenchmarker-fake-package>=1.2
+    """
+    with self._MockOpen(requirements_content):
+      with self.assertRaises(errors.Setup.PythonPackageRequirementUnfulfilled):
+        requirements._CheckRequirements(_PATH)
+
+  def testIncorrectVersion(self):
+    requirements_content = """
+    # The version of the installed python-gflags will be less than 42.
+    python-gflags>=42
+    """
+    with self._MockOpen(requirements_content):
+      with self.assertRaises(errors.Setup.PythonPackageRequirementUnfulfilled):
+        requirements._CheckRequirements(_PATH)
+
```

Comment:
Can you add a test case for when a user has an old version of the package installed but the requirements specify a specific newer version, and another test case for the reverse, that is, the user has a newer version of the package but an older version is required?

**Example for test case 1:** - this one is probably covered by `testIncorrectVersion()`

User has old version of the package:
python-novaclient==2.15.0

but requirements ask for:
python-novaclient==2.35.0


**Example for test case 2:**

User has newer version of package:
python-novaclient==3.2.0

Requirements ask for:
python-novaclient==2.35.0